### ISSUE TITLE
Multiple AWS::Serverless::Api only yields single record set

### ISSUE THREAD

**Issue by Matthew-Beckett** [aws/serverless-application-model#1829]
### Description:

When creating multiple AWS::Serverless::Api resources, with Domain and Route53 properties, only one gateway has a record set created when rendered to CloudFormation template from SAM.

```yaml
  ApiGatewayAdmin:
    Type: AWS::Serverless::Api
    Properties:
      Name: !Join ["-", ["App", !Ref EnvironmentName, "Web"]]
      StageName: !Ref EnvironmentName
      TracingEnabled: true
      MethodSettings:
        - LoggingLevel: Info
          ResourcePath: '/*'
          HttpMethod: '*'
      Domain:
        DomainName: !Join [".", ["admin", !Ref EnvironmentName, !Ref DomainName]]
        CertificateArn: !Ref Certificate
        EndpointConfiguration: REGIONAL
        Route53:
          HostedZoneId: !Ref HostedZone
      EndpointConfiguration:
        Type: REGIONAL
```

I have three objects exactly identical as above with only changes to the DomainName to make it unique and possibly co-incidentally, the API which successfully has its record set created is the final API object in the YAML before function definitions.

### Steps to reproduce:

- Create three API resources, with DomainName and Route53 configuration sharing the same hosted zone
- Render template to CloudFormation

### Observed result:

CloudFormation template output only contains a single record set, in my case, the last defined API resource.

### Expected result:

A record set and domain name for each API resource

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: MacOS, Linux (Alpine, Ubuntu)
2. `sam --version`: SAM CLI, version 1.2.0

`Add --debug flag to command you are running`

**Comment by CoshUS** [comment]
I was able to reproduce it with sam-translator 1.30.0.

template.yaml
```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: >
  apigateway-2402

  Sample SAM Template for apigateway-2402

Resources:
            
  ApiGatewayAdminOne:
    Type: AWS::Serverless::Api
    Properties:
      Name: App-Prod-Web
      StageName: Prod
      TracingEnabled: true
      MethodSettings:
        - LoggingLevel: Info
          ResourcePath: '/*'
          HttpMethod: '*'
      Domain:
        DomainName: admin.one.amazon.com
        CertificateArn: arn::cert::abc
        EndpointConfiguration: REGIONAL
        Route53:
          HostedZoneId: "abc123456"
      EndpointConfiguration:
        Type: REGIONAL

            
  ApiGatewayAdminTwo:
    Type: AWS::Serverless::Api
    Properties:
      Name: App-Prod-Web
      StageName: Prod
      TracingEnabled: true
      MethodSettings:
        - LoggingLevel: Info
          ResourcePath: '/*'
          HttpMethod: '*'
      Domain:
        DomainName: admin.two.amazon.com
        CertificateArn: arn::cert::abc
        EndpointConfiguration: REGIONAL
        Route53:
          HostedZoneId: "abc123456"
      EndpointConfiguration:
        Type: REGIONAL
        
            
  ApiGatewayAdminThree:
    Type: AWS::Serverless::Api
    Properties:
      Name: App-Prod-Web
      StageName: Prod
      TracingEnabled: true
      MethodSettings:
        - LoggingLevel: Info
          ResourcePath: '/*'
          HttpMethod: '*'
      Domain:
        DomainName: admin.three.amazon.com
        CertificateArn: arn::cert::abc
        EndpointConfiguration: REGIONAL
        Route53:
          HostedZoneId: "abc123456"
      EndpointConfiguration:
        Type: REGIONAL
```

translated.yaml
```
...
    "RecordSetGroup370194ff6e": {
      "Type": "AWS::Route53::RecordSetGroup",
      "Properties": {
        "HostedZoneId": "abc123456",
        "RecordSets": [
          {
            "Name": "admin.three.amazon.com",
            "Type": "A",
            "AliasTarget": {
              "HostedZoneId": {
                "Fn::GetAtt": [
                  "ApiGatewayDomainName41bfc7f9c4",
                  "RegionalHostedZoneId"
                ]
              },
              "DNSName": {
                "Fn::GetAtt": [
                  "ApiGatewayDomainName41bfc7f9c4",
                  "RegionalDomainName"
                ]
              }
            }
          }
        ]
      }
    },
...
```

**Comment by alexandrenicol** [comment]
Also experiencing the same behaviour: 2 AWS::Serverless::Api in 1 template --> 1 RecordSet.

**Comment by aaythapa** [comment]
Closing this as the issue was fixed with #2408
